lib/analyzer/image-extractor.ts

```diff
@@ -1,110 +1,181 @@
       resolve(result);
     });
-    stream.pipe(layerExtract);
+    layerTarStream.pipe(layerExtract);
   });
 }
 
 /**
- * Extract key files textual content and MD5 sum from the specified TAR file.
+ * Retrieve the products of files content from the specified TAR file.
  * @param imageTarPath path to image file saved in tar format
- * @param txtPatterns list of plain text key files paths patterns to extract and return as strings
- * @param md5Patterns list of binary key files paths patterns to extract and return their MD5 sum
- * @returns manifest file and key files by inner layers
+ * @param searchActions array of pattern, callbacks pairs
+ * @returns array of extracted files products sorted by the reverse order of
+ *  the layers from last to first
  */
-async function extractImageKeyFiles(
+async function extractLayersFromTar(
   imageTarPath: string,
-  txtPatterns: string[],
-  md5Patterns: string[] = [],
-): Promise<ExtractedImage> {
+  searchActions: SearchAction[],
+): Promise<SearchActionProducts[]> {
   return new Promise((resolve) => {
     const imageExtract: Extract = extract();
-    let manifest: string;
-    const layers: Layers = {};
+    const layers: { [key: string]: SearchActionProducts } = {};
+    let layersNames: string[];
 
-    imageExtract.on("entry", (header, stream, next) => {
+    imageExtract.on("entry", async (header, stream, next) => {
       if (header.type === "file") {
         if (basename(header.name) === "layer.tar") {
-          extractLayerKeyFiles(stream, txtPatterns, md5Patterns).then(
-            (extractedKeyFiles) => {
-              layers[header.name] = extractedKeyFiles;
-            },
-          );
+          layers[header.name] = await extractFromLayer(stream, searchActions);
         } else if (header.name === "manifest.json") {
           streamToString(stream).then((manifestFile) => {
-            manifest = manifestFile;
+            const manifest = JSON.parse(manifestFile);
+            layersNames = manifest[0].Layers;
           });
         }
       }
       stream.resume(); // auto drain the stream
       next(); // ready for next entry
     });
     imageExtract.on("finish", () => {
-      // all image level entries read
-      resolve({
-        manifest,
-        layers,
-      });
+      // reverse layers order from last to first
+      // skip (ignore) non-existent layers
+      // return the layers content without the layer name
+      resolve(
+        layersNames
+          .reverse()
+          .filter((layersName) => layers[layersName])
+          .map((layerName) => layers[layerName]),
+      );
     });
     createReadStream(imageTarPath).pipe(imageExtract);
   });
 }
+
+/**
+ * Extract key files textual content and MD5 sum from the specified TAR file
+ * @param imageTarPath path to image file saved in tar format
+ * @param searchActions array of pattern, callbacks pairs
+ * @returns extracted files products
+ */
+async function extractFromTar(
```

Comment:
please revert the changes of this function so it will be like before (https://github.com/snyk/snyk-docker-plugin/pull/89/files#diff-1a778f82722b9f884412efd52d3c46d9L157) - I don't want this change as part of this PR